src/amd/compiler/aco_instruction_selection.cpp

```diff
@@ -2848,16 +2848,26 @@ void visit_image_load(isel_context *ctx, nir_intrinsic_instr *instr)
    const nir_variable *var = nir_deref_instr_get_variable(nir_instr_as_deref(instr->src[0].ssa->parent_instr));
    const struct glsl_type *type = glsl_without_array(var->type);
    const enum glsl_sampler_dim dim = glsl_get_sampler_dim(type);
+   Temp dst = get_ssa_temp(ctx, &instr->dest.ssa);
 
    if (dim == GLSL_SAMPLER_DIM_BUF) {
-      unreachable("image load with GLSL_SAMPLER_DIM_BUF not yet implemented\n");
+      Temp rsrc = get_sampler_desc(ctx, nir_instr_as_deref(instr->src[0].ssa->parent_instr), ACO_DESC_BUFFER, nullptr, true, true);
+      Temp vindex = emit_extract_vector(ctx, get_ssa_temp(ctx, instr->src[1].ssa), 0, v1);
+
+      aco_ptr<MUBUF_instruction> load{create_instruction<MUBUF_instruction>(aco_opcode::buffer_load_format_xyzw, Format::MUBUF, 3, 1)};
```

Comment:
is that always _xyzw? or do we have the possibilty to shrink that?
does nir_opt_shrink_loads() do that or do we have to do that on our own?

Reply:
I think there is the possibility to shrink it.
I think I had a TODO for it but it seems it got lost at some point.
I'll work on doing that